### PR TITLE
feat: tips setting

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MF-1518

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/icons/general/Direct.svg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 36 36">
```
Comment (UncleBill) [inline packages/icons/general/Direct.svg]: Use `Icons.SimpleEmpty`

File: packages/mask/src/plugins/Tips/SNSAdaptor/TipsEntranceDialog.tsx [hunk 2]
```diff
@@ -59,232 +62,154 @@ const useStyles = makeStyles()((theme) => ({
...
 
-    const [pendingDefault, setPendingDefault] = useState<string>()
-    const defaultAddress = bindingWallets.find((x) => x.isDefault)?.identity
+    // Sort By `last_checked_at`
+    const bindingWallets = useMemo(() => {
+        if (!proofRes?.length) return EMPTY_LIST
+        return sortBy(proofRes, (x) => -Number.parseInt(x.last_checked_at, 10)).map(
+            (wallet, index, list): BindingProof => ({
+                ...wallet,
+                rawIdx: list.length - index - 1,
```
Comment (UncleBill) [inline packages/mask/src/plugins/Tips/SNSAdaptor/TipsEntranceDialog.tsx]: Looks like `rawIdx` only for `fallbackName`, let's just construct it in the initialization.
```suggestion
                fallbackName: `Wallet ${list.length - index - 1}`,
```

```tsx
                        <WalletItem
                            key={x.identity}
                            setAsDefault={setAsDefault}
                            fallbackName={`Wallet ${x.rawIdx}`}
                            address={x.identity}
                            isDefault={isSameAddress(defaultAddress, x.identity)}
                        />
```

File: packages/mask/src/plugins/Tips/SNSAdaptor/TipsSetting.tsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+    const [addresses, setAddresses] = useState<string[]>([])
+
+    const { showSnackbar } = useCustomSnackbar()
+
+    const { value: hiddenAddress, loading } = useHiddenAddressSetting(PluginId.Tips, currentPersona?.publicKeyAsHex)
+
+    const onSwitchChange = useCallback((address: string) => {
+        setAddresses((prev) => {
+            return prev.includes(address) ? prev.filter((x) => !isSameAddress(address, x)) : [...prev, address]
```
Comment (UncleBill) [inline packages/mask/src/plugins/Tips/SNSAdaptor/TipsSetting.tsx]:
```suggestion
            return prev.some( x => isSameAddress(address, x)) ? prev.filter((x) => !isSameAddress(address, x)) : [...prev, address]
```

We never change `address` throughout the process, I think it's safe to compare string directly.

```suggestion
            return prev.includes(address) ? prev.filter((x) => address !== x) : [...prev, address]
```

File: packages/mask/src/plugins/Tips/hooks/usePublicWallets.ts [hunk 4]
```diff
@@ -24,30 +24,17 @@ export function usePublicWallets(persona: ECKeyIdentifier | undefined): TipAccou
...
-        const storage = Storage.createNextIDStorage(
-            persona.publicKeyAsHex,
-            NextIDPlatform.NextID,
-            persona.publicKeyAsHex,
-        )
-        const wallets = (await storage.get<BindingProof[]>(PluginId.Tips))?.filter(
-            (x) => x.platform === NextIDPlatform.Ethereum,
+        return (
+            nextIdWallets
+                ?.filter((x) => !TipsSetting?.hiddenAddresses?.some((y) => isSameAddress(y, x.address)) ?? true)
```
Comment (UncleBill) [inline packages/mask/src/plugins/Tips/hooks/usePublicWallets.ts]: You might change this code bellow as well.


```ts
    return useMemo(() => {
        // If user configured wallets for tips, it couldn't be null.
        return walletsFromCloud || uniqBy(nextIdWallets || [], (x) => x.address)
    }, [nextIdWallets, walletsFromCloud])
```

File: packages/icons/general/Direct.svg (hunk 1 above)
Comment (nuanyang233) [reply packages/icons/general/Direct.svg]: Do u means rename this svg file to `SimpleEmpty`?

File: packages/mask/src/plugins/Tips/SNSAdaptor/TipsEntranceDialog.tsx [hunk 5]
```diff
@@ -293,63 +215,59 @@ export function TipsEntranceDialog({ open, onClose }: TipsEntranceDialogProps) {
...
-                        {t.cancel()}
-                    </ActionButton>
-                    <ActionButton loading={kvFetchState.loading} fullWidth disabled={!isDirty} onClick={onConfirm}>
+            <DialogActions className={classes.actions}>
+                <div className={classes.personaInfo}>
+                    {currentPersona?.avatar ? (
+                        <Avatar src={currentPersona.avatar} className={classes.avatar} />
+                    ) : (
+                        <Icons.MenuPersonasActive
+                            color="#f9fafa"
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Tips/SNSAdaptor/TipsEntranceDialog.tsx]: Use a themed color.

File: packages/shared/src/UI/components/WalletSettingCard/UI.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+        },
+    },
+    '& .MuiSwitch-thumb': {
+        boxSizing: 'border-box',
+        width: 14,
+        height: 14,
+    },
+    '& .MuiSwitch-track': {
+        borderRadius: 26 / 2,
+        backgroundColor: theme.palette.mode === 'light' ? '#E9E9EA' : '#39393D',
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/WalletSettingCard/UI.tsx]: Use themed colors.

File: packages/icons/general/Direct.svg (hunk 1 above)
Comment (UncleBill) [reply packages/icons/general/Direct.svg]: We can remove this file now.

**Review by UncleBill — DISMISSED** [review]

**Review by guanbinrui — DISMISSED** [review]

**Pull request merged**